### PR TITLE
Site Editor: Only mark the 'Site' menu item active when editing a home template

### LINKED ISSUE #36821: Site Editor: "Site" menu item is active when editing a template
Screenshot 2021-11-24 at 14 35 38

The "Site" menu item should not be active in this case.

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Fixes #36821.

PR marks the "Site" menu item active only when editing a home template.

## Why?
See #36821.

## How?
Compares if template ID from navigation and editor settings match.

## Testing Instructions
1. Open Site Editor.
2. Confirm that the "Site" menu item is active.
3. Open different page templates (Single, Page) in Editor.
4. Confirm that the "Site" menu isn't active.

## Screenshots or screencast 
CleanShot 2022-07-29 at 13 20 06

### CODE REVIEW

**Comment by jameskoster** [comment]
Thanks for taking this on @Mamaduka :)

One problem with this execution is that if you have a static homepage (which uses the Page template), then the 'Site' menu item appears active if you open the Page template which feels a bit strange: 

Screenshot 2022-07-29 at 11 53 01

Ideally 'Site' should **only** be active when:

1. You click "Appearance > Editor"
2. You click the Site menu item itself

The issue on trunk is that Site also becomes active when you select a template.

**Comment by Mamaduka** [comment]
@jameskoster, I'm not able to reproduce the issue. For example, if I visit `twentytwentytwo//page`, the Site menu item isn't active.

P.S. The homepage will say "Page" in the top toolbar when using a static template.

**Comment by jameskoster** [comment]
Hmm, you're right. Maybe my build didn't complete before I refreshed 🙈 Seems to be working now :) 

I did notice another issue (although perhaps not something to fix here). If you click 'Site' from the templates list it will collapse the nav panel and load the site homepage. However if you click 'Site' whilst editing a template then the nav panel remains open. I wonder if these behaviours should be consistent? Happy to open a new issue if you prefer :)

**Comment by Mamaduka** [comment]
@jameskoster, I believe that's intentional and not related to these changes.

https://github.com/WordPress/gutenberg/blob/5bfc80b8af118fcaaa32336a5281ff1fcf94c6c6/packages/edit-site/src/components/app/index.js#L51-L59

**Comment by jameskoster** [comment]
@Mamaduka thanks for looking into that, I guess we can address that point separately – I'll open an issue. 

Returning to this PR, I see only one remaining scenario to consider – the front-page template. Since that will always resolve to display the homepage regardless of reading settings, perhaps we should highlight the 'Site' menu item when it's being edited. What do you think?

**Comment by Mamaduka** [comment]
@jameskoster, that's a good point. I've updated the logic to account for `front-page`.

P.S. I've also realized that I made a mistake in the `_resolve_home_block_template` logic 😅  I will try to fix that as well.

File: packages/edit-site/src/components/app/index.js
Comment (Mamaduka) [inline packages/edit-site/src/components/app/index.js]: @kevin940726, do you think it's worth extracting this into a utility function like `getIsListPage`?

Comment (kevin940726) [reply packages/edit-site/src/components/app/index.js]: Oops, only just saw this now. I don't have a preference, but usually I'd only do that if it's repeated multiple times.

**Review by kevin940726 — APPROVED** [review]
LGTM 🧡

File: packages/edit-site/src/components/app/index.js
Comment (Mamaduka) [reply packages/edit-site/src/components/app/index.js]: I do the same. I will extract it into a function when there's a need.

**Comment by Mamaduka** [comment]
I rebased and resolved merge conflicts. I think the remaining question is about a11y.

To quote @kevin940726 - https://github.com/WordPress/gutenberg/issues/36821#issuecomment-981497454
> The problem is that we do a redirection upon clicking on the "Site" link. There's essentially no page which is called "Site", we immediately redirect the user to the template editor (which is the frontpage template by default). Should we:
> 
> * Highlight the "Site" link only when we're on the default(frontpage) template, or
> * Highlight the "Site" link whenever we're editing any template but not template parts,
> * Or anything else
> 
> Since we always do a full navigation, so we don't have the information of where we're coming from (whether we click on the "Site" link or not). Since this would likely be linked to the `aria-current` attribute in #36946, I guess it makes sense to have accessibility folks giving feedbacks on this as well. WDYT?

**Comment by jameskoster** [comment]
I don't think I understand the problem 🤔

**Comment by Mamaduka** [comment]
@jameskoster, If I understand correctly, the problem is missing `aria-current` in the navigation when a user is editing the "non-home template." That couldn't be problematic for a11y.

**Comment by jameskoster** [comment]
It seems justified to me that a navigation menu can entertain a state where there is no currently-selected item. So I don't think that should be a problem. It would be good to get a11y sign-off from someone with more expertise though :D

**Comment by Mamaduka** [comment]
@jameskoster, should we merge this? Will the same active menu logic apply once the Browser Mode is merged?

**Comment by jameskoster** [comment]
It's not yet clear whether the 'Site' or 'Browse' menu item will be needed long term (https://github.com/WordPress/gutenberg/pull/44770 for context). But I don't think that should stop us merging this PR, since it's a clear enhancement on the current behaviour.

**Comment by Mamaduka** [comment]
Thanks, @jameskoster 🙇

I just rebased and will merge after all checks are green. We'll probably get more feedback after this is merged into the plugin.

**Pull request merged**
